site/src/pages/release-notes/1.24.0.mdx

```diff
@@ -0,0 +1,279 @@
+ 
+  </Tip>
+- **Retrieving CallOptions with GrpcCallOptions**: With GrpcCallOptions in Armeria, you can now easily retrieve
+  `CallOptions` from a given <type://RequestContext>.
+  #4776 #4822
+  ```java
+  GrpcClients
+    .builder(grpcServerUri)
+    .decorator((delegate, ctx, req) -> {
+      CallOptions options = GrpcCallOptions.get(ctx);
+      MyOption myOption = options.getOption(myOptionKey);
+      // act on myOption if needed
+      ...
+  
+      return delegate.execute(ctx, req);
+    })
+    .build(MyGrpcStub.class);
+  ```
+- **Fallback Strategy for CircuitBreakerClient**: You can now easily set a fallback strategy to 
+  <type://CircuitBreakerClient> when a <type://CircuitBreaker> is open. #4800 #4818
+  ```java
+  CircuitBreakerClient
+    .builder(...)
+    .recover((ctx, req) -> {
+      // A fallback response when the circuit is open.
+      return HttpResponse.of(...);
+    });
+  ```
+- **Netty ChannelPipeline Customizer**: You can now customize the Netty `ChannelPipeline` for the server side #4710 #4813
+  ```java
+  Server
+    .builder()
+    .childChannelPipelineCustomizer(pipeline -> {
+      // Add a custom handler to the pipeline.
+    })
+    .builder();
+  ```
+- **HTTP/2 over TLS without ALPN**: Armeria now supports using HTTP/2 over TLS without the need for 
+  [ALPN](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-http2-17#section-3.3) #4702 #4704
+  ```java
+  ClientFactory
+    .builder()
+    .useHttp2WithoutAlpn(true)
+    ...
+    .build();
+  ```
+- **Armeria with Jetty**: Armeria is now compatible with Jetty 10 and Jetty 11 #4845
+- **Armeria with Thrift**: Armeria is now compatible with Thrift 0.18.1. #4812
+
+## 📈 Improvements
+
+- `RequestContext.makeContextAware()` now returns instances of <type://ContextAwareRunnable>,
+  <type://ContextAwareCallable>, <type://ContextAwareFunction>, <type://ContextAwareBiFunction>,
+  <type://ContextAwareConsumer> and <type://ContextAwareBiConsumer>. #4878 #4890
+- Improved annotated services now perform as fast as functional-style services under the following conditions: #4888
+  - The annotated method returns an `HttpResponse`; and
+  - The annotated method is run on the event loop (i.e. not annotated with `@Blocking`).
+- You can now use the following features with Spring Boot Actuator integration. #4874 #4880
+  - `management.endpoints.web.exposure.*`
+  - `management.endpoints.web.path-mapping.*`
+  - `management.endpoints.web.discovery.*`
+  - `management.endpoint.health.status.http-mapping.*`
+  - `application/vnd.spring-boot.actuator.v3+json`, `application/vnd.spring-boot.actuator.v2+json` and
+    `application/json`
+- You can now correctly set the default log name for <type://VirtualHost> via
+  <type://VirtualHostBuilder#defaultLogName(String)> #4817
+- Armeria client now exports metrics for <type://RetryingClient> with the following metric names: #3350 #4707
+  - `armeria.client.actual.requests`
```

Comment:
nit; I think this metric already existed before

Reply:
Oops.